ctapipe/irf/irf_classes.py

```diff
@@ -0,0 +1,74 @@
+"""
+Define a parent IrfTool class to hold all the options
+"""
+from enum import Enum
+
+import astropy.units as u
+from pyirf.cuts import calculate_percentile_cut
+from pyirf.spectral import CRAB_HEGRA, IRFDOC_ELECTRON_SPECTRUM, IRFDOC_PROTON_SPECTRUM
+
+from ..core import Component
+from ..core.traits import Float, Integer
+
+
+class Spectra(Enum):
+    CRAB_HEGRA = 1
+    IRFDOC_ELECTRON_SPECTRUM = 2
+    IRFDOC_PROTON_SPECTRUM = 3
+
+
+PYIRF_SPECTRA = {
+    Spectra.CRAB_HEGRA: CRAB_HEGRA,
+    Spectra.IRFDOC_ELECTRON_SPECTRUM: IRFDOC_ELECTRON_SPECTRUM,
+    Spectra.IRFDOC_PROTON_SPECTRUM: IRFDOC_PROTON_SPECTRUM,
+}
+
+
+class ThetaCutsCalculator(Component):
+    theta_min_angle = Float(
+        default_value=-1, help="Smallest angular cut value allowed (-1 means no cut)"
+    ).tag(config=True)
+
+    theta_max_angle = Float(
+        default_value=0.32, help="Largest angular cut value allowed"
+    ).tag(config=True)
+
+    theta_min_counts = Integer(
+        default_value=10,
+        help="Minimum number of events in a bin to attempt to find a cut value",
+    ).tag(config=True)
+
+    theta_fill_value = Float(
+        default_value=0.32, help="Angular cut value used for bins with too few events"
+    ).tag(config=True)
+
+    theta_smoothing = Float(
+        default_value=-1,
```

Comment:
What does `-1` mean here? If that is supposed to mean "option not given, do not apply smoothing", please use `default_value=None, allow_none=True`.